ffi/src/clipboard/windows.rs

```diff
@@ -0,0 +1,134 @@
+#![allow(clippy::unused_self)] // We want to keep the signature of the function stay the same between windows and non-windows
+
+use super::ffi::CliprdrBackendFactory;
+use crate::error::ffi::IronRdpError;
+#[cfg(not(windows))]
+use crate::error::WrongOSError;
+#[cfg(not(windows))]
+use ironrdp_cliprdr_native as _; // avoid linter error, stub clipboard will be used in later commit
+
+/*
+    Why are we creating a WinCliprdrInner struct and implement differently?
+
+    1. We want to keep the FFI interface align with generated bindings.
+    2. ironrdp_cliprdr_native::WinClipboard only compiles if the target is windows.
+    3. We do not want to put any conditional compilation in the ffi module.
+
+    Hence we create a WinCliprdrInner struct and implement it differently based on the target platform.
+    and throw WrongOSError if the target platform is not windows.
+*/
+#[diplomat::bridge]
+pub mod ffi {
+
+    use crate::clipboard::ffi::CliprdrBackendFactory;
+    use crate::clipboard::message::ffi::ClipboardMessage;
+    use crate::error::ffi::IronRdpError;
+    use crate::error::{PointWidth, WrongPointWidthError};
+
+    use super::WinCliprdrInner;
+
+    #[diplomat::opaque]
+    pub struct WinCliprdr(WinCliprdrInner);
+
+    impl WinCliprdr {
+        pub fn new_32bit(hwnd: u32) -> Result<Box<WinCliprdr>, Box<IronRdpError>> {
```

Comment:
could we keep passing `hwnd` as `isize` here instead of two separate methods for 32/64 bits?

Reply:
it actually can be isize ! I checked C# documentation and diplomat source code, yes, we don't have to worry about the pointer  width anymore.

Reply:
updated